### PR TITLE
Adding basic file configuration options

### PR DESCRIPTION
**Pull request by jhadvig**
We will now generate the configuration file for every run of mongodb, using environment variables as input. Substitution is done with envsubst.

### CODE REVIEW

**Comment by jhadvig** [comment]
@mnagy @mfojtik PTAL

**Comment by mfojtik** [comment]
LGTM

File: 2.4/run-mongod.sh [hunk 1]
```diff
@@ -5,17 +5,33 @@ source /var/lib/mongodb/.bashrc
...
 
+# Data directory where MongoDB database files live. The data subdirectory is here
+# because .bashrc and mongodb.conf both live in /var/lib/mongodb/ and we don't want a
+# volume to override it.
+export MONGODB_DATADIR=/var/lib/mongodb/data
+
+# Configuration settings.
+export MONGODB_NOPREALLOC=${MONGODB_NOPREALLOC:-true}
+export MONGODB_SMALLFILES=${MONGODB_SMALLFILES:-true}
+export MONGODB_QUITE=${MONGODB_QUITE:-true}
```
Comment (mnagy) [inline 2.4/run-mongod.sh]: The word "quiet" is not quite correct.

File: README.md [hunk 2]
```diff
@@ -52,13 +52,23 @@ Choose between CentOS7 or RHEL7 based image:
...
+|  `MONGODB_DATABASE`       | Database name (optional)                    |
 |  `MONGODB_ADMIN_PASSWORD` | Password for the admin user (optional)      |
 
+
+Following environment variables influence MongoDB configuration file. They are all optional.
+
+|    Variable name      |    Description                                                            |    Default
+| :-------------------- | ------------------------------------------------------------------------- | ----------------
+|  `MONGODB_NOPREALLOC` | Disable data file preallocation.                                          |  true
+|  `MONGODB_SMALLFILES` | Set MongoDB to use a smaller default data file size                       |  true
```
Comment (mnagy) [inline README.md]: Nit: Missing period.

**Comment by jhadvig** [comment]
@mnagy PTAL

**Comment by mnagy** [comment]
LGTM

**Pull request merged**